### PR TITLE
Vim improvements

### PR DESCRIPTION
**Pull request by jesse-wei**
This PR adds two Vim plugins (strict improvement): auto-pairs and easymotion. It also modifies two settings: relative numbering (opinionated change) and smartcase.

If this PR and https://github.com/comp211/learncli211/pull/8 are merged, then users can edit `/mnt/learncli/.vimrc` to have changes persist across sessions. This is necessary this semester because students may not like relative line numbering, which is made default in this PR.

### Plugins (strict improvements, can skip over this section)

Auto-pairs automatically closes `(`, `[` and `{` with the corresponding right char with correct indentation when the left char is typed. It functions the same as any IDE, as far as I can tell. Without the plugin, users have to manually type and indent all right chars.

EasyMotion allows for easy motion in files. In this config, it is used with Space + Space + <action>. For example, Space + Space  + w starting from the `m` in `int main` results in something like

image

Then pressing `c` would put the cursor on the `return`, `o` on the `printf`, etc.

Space + s and Space + t are mapped to EasyMotion's 2-character search feature. The keybind mappings don't override any existing ones since I use the leader key Space as the prefix.

### Settings (opinionated but good (imo) changes)

Relative numbering is an opinionated change, but one that is used by many Vim users. It looks like this (cursor is on absolute line 7):

image

The point is that in Vim, users are meant to use, e.g., `5j` to move down 5 lines. With relative numbering, you can see that the `printf` is at 5 so you know to type `5j`. However, this is inconvenient with absolute line numbers because you have to compute `{12-7}j` (inconvenient when line numbers are large). Same holds for similar motions such as `{num}dd` as well.

When you do need absolute line number (e.g., "go to line 7"), the absolute line number of the current line is still shown, and you can still easily navigate to any absolute line using `{line}G`.

I believe this setting makes Vim much easier to use, but I understand if this can't be accepted into the repo because it is not normally used in other IDE's and might cause some student questions.

The other setting change is pretty much a strict improvement. Without ignorecase+smartcase, in the above file, `/h` to find the `H` in `"Hello world"` results in nothing found because `H` is capitalized. This behavior is not like that of normal IDE's. With ignorecase+smartcase, `/The` would find only `"The"`, while `/the` would find `"the"` or `"The"`, etc.

### CODE REVIEW

**Comment by jesse-wei** [comment]
Requesting reviews from @donporter and @brent-munsell since this will affect student Vim usage in your courses.

**Comment by eric-unc** [comment]
Relative line numbers are supported in other IDEs, but arguably not super useful outside of Vim. I love them but unfortunately I think students might be very confused.

**Comment by jesse-wei** [comment]
> Relative line numbers are supported in other IDEs, but arguably not super useful outside of Vim. I love them but unfortunately I think students might be very confused.

Understandable, Brent said he's okay with relative line numbering being the default, and if a student doesn't like it, they can put `sed -i '/relativenumber/d' /root/.vimrc` in their `/mnt/learncli/.bash_profile`.

For convenience, `echo "sed -i '/relativenumber/d' /root/.vimrc" >> /mnt/learncli/.bash_profile`.

**Comment by eric-unc** [comment]
Okay sounds good

**Comment by jesse-wei** [comment]
> > Relative line numbers are supported in other IDEs, but arguably not super useful outside of Vim. I love them but unfortunately I think students might be very confused.
> 
> Understandable, Brent said he's okay with relative line numbering being the default, and if a student doesn't like it, they can put `sed -i '/relativenumber/d' /root/.vimrc` in their `/mnt/learncli/.bash_profile`.
> 
> For convenience, `echo "sed -i '/relativenumber/d' /root/.vimrc" >> /mnt/learncli/.bash_profile`

Update to this, if https://github.com/comp211/learncli211/pull/8 is merged, students can `git pull` that repo so that `/mnt/learncli/.vimrc` exists, then edit that file to comment out the line `set relativenumber`.

**Comment by jesse-wei** [comment]
@eric-unc Brent approves and all 4 PR's (the other PR in this repo is #29, and there are two at learncli211: https://github.com/comp211/learncli211/pull/7 and https://github.com/comp211/learncli211/pull/8) are ready for merge. Can you approve so I can fix any merge conflicts and merge?

File: .bashrc [hunk 1]
```diff
@@ -156,4 +156,13 @@ if [ ! -d "{$hostdir}/bin" ]
...
\ No newline at end of file
+export PATH="${PATH}:${hostdir}/bin"
+
+# To allow persistent vimrc modifications,
+# copy from .vimrc on host to ~/.vimrc
+# and install plugins in subshell in background, suppressing output
+hostvimrc="${hostdir}/.vimrc"
+if [ -f "$hostvimrc" ]; then
+    cp "$hostvimrc" ~/.vimrc
+    (vim -c -es 'PluginInstall' -c -es 'qa!' </dev/null >/dev/null 2>&1 &)
```
Comment (eric-unc) [inline .bashrc:167]: nit: spaces

**Review by eric-unc — APPROVED** [review]
LGTM

File: .bashrc (hunk 1 above)
Comment (jesse-wei) [reply .bashrc:167]: iirc, if I make a commit fixing this you'll have to re-approve, so I'll merge this now and fix later (TM).

**Pull request merged**